pkg/platform/api/mono/mono.go

```diff
@@ -34,7 +37,32 @@ func Init(serviceURL *url.URL, auth *runtime.ClientAuthInfoWriter) *mono_client.
 	if auth != nil {
 		transportRuntime.DefaultAuthentication = *auth
 	}
-	return mono_client.New(transportRuntime, strfmt.Default)
+	client := mono_client.New(transportRuntime, strfmt.Default)
+
+	// For the Oauth client, prefer use of IPv4. This is needed particularly for device
+	// authorization, which compares the IP address of the State Tool request and the IP address of
+	// the web client request. If there's a mismatch, authorization fails. When this happens, it's
+	// often because the State Tool connects to the Platform via IPv6, but the browser does via IPv4
+	// (browsers apparently prefer IPv4 for now).
+
+	httpTransport, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		return client // for some reason the assertion can fail...
+	}
+
+	ipv4TransportRuntime := httptransport.New(serviceURL.Host, serviceURL.Path, []string{serviceURL.Scheme})
```

Comment:
Since this can also return an ipv6 transport I don't think we should call it `ipv4`, unless we call it something like `ipv4Preferred`.

Reply:
Fair point.